amethyst_animation/src/resources.rs

```diff
@@ -499,7 +499,7 @@ where
             .iter_mut()
             .filter(|t| t.control_id == control_id)
             .filter(|t| t.state != ControlState::Done)
-            .map(|c| (samplers.get(&c.sampler).unwrap(), c))
+            .filter_map(|c| samplers.get(&c.sampler).map(|s| (s, c)))
```

Comment:
I'm not super familiar with the animation crate, but is it really better to silently fail here?

Reply:
No, this should probably fail hard, because if you're referring to a missing sampler you're in deep shit

Reply:
@Rhuagh should I also unwrap [here](https://github.com/amethyst/amethyst/blob/master/amethyst_animation/src/systems/sampling.rs#L65)?